### ISSUE TITLE
Reconsider descriptions referring to density

### ISSUE THREAD

**Issue by terracoda** [phetsims/gravity-force-lab-basics#149]
Based on comments from @arouinfar's https://github.com/phetsims/gravity-force-lab-basics/issues/142#issue, we may need to adjust how we are describing density in help text and alerts.

**Note** @arouinfar, I said that you did not experience the latest version of the sim's descriptions, but in fact you may have. Did you use this link from the Dev Test issue:
 https://phet-dev.colorado.edu/html/gravity-force-lab-basics/1.0.0-dev.39/phet/gravity-force-lab-basics_all_phet.html

Now quoting from your comment https://github.com/phetsims/gravity-force-lab-basics/issues/142#issuecomment-501894516
> As for the description content itself, I have one area of concern. When Constant Size is checked, the help text under H3: Mass Controls reads:

> Change {{density of spheres}} in billions of kilograms.

> Density is measured in mass per unit volume, so "billions of kilograms" is incorrect. It would more accurate to say "billions of kilograms per cubic kilometer" or even the more general "billions of kilograms per unit volume".

> That said, I have a bigger question as to whether this help text and alerts ("As {{density}} decreases, force arrows get smaller..) should refer to densities at all. Yes, the density will change, but that's as a result of the user changing the value of the mass. The goal here is for students to understand the relationship between the mass and force. When constant size is unchecked, some students may conflate mass and size, so the Constant Size option allows students to really just focus on the mass itself. I think changing the descriptions to refer to density adds on the layer of complexity we were trying to eliminate in the first place.

There are 2 parts to this issue regarding descriptions when the Constant Size checkbox is checked:
1. What help text do we want?
2. Do we need a change to the phrasing of the current alert?

**Comment by terracoda** [comment]
**Part 1 of issue, the help text:**
Please consider all descriptions that are trying to support the idea of density without literally describing how density is represented visually (i.e., with shades of Red and Blue). 

**Note** that when a user checks or unchecks the Constant Size checkbox, there are two different alerts:
- Upon checking Constant Size: Mass 1 and mass 2 set to same size.
- Upon UNchecking Constant Size, the alert is dynamic based on current values for mass: Mass 1 {{half the size of}} mass 2.

So I am wondering if you prefer that we not change the help text at all (version 2) when Constant Size is checked, or should we use the more accurate description with the unit of volume added (version 3 and 4):
1. Current help text (both states):
  - Constant Size is NOT checked: Change {{mass of spheres}} in billions of kilograms.
  - Constant Size is checked: Change {{density of spheres}} in billions of kilograms. 
2. No change in help text:
  - Constant Size is NOT checked: Change mass of spheres in billions of kilograms.
  - Constant Size is checked: Change mass of spheres in billions of kilograms. 
3. Density and exact volume when Constant Size checked :
- Constant Size is NOT checked: Change {{mass of spheres}} in billions of kilograms.
- Constant Size is checked: Change {{density of spheres}} in billions of kilograms {{per cubic kilometer}}. 
4. Density and general volume when Constant Size checked:
- Constant Size is NOT checked: Change {{mass of spheres}} in billions of kilograms.
- Constant Size is checked: Change {{density of spheres}} in billions of kilograms {{per unit volume}}. 

@arouinfar, which of these pairs is your preference for a middle-school sim? Keep in mind, if for Gravity Force Lab (regular) the help text needs to be more precise, the help text can be different between the two sims.

**Comment by terracoda** [comment]
## Part 2 of issue, the alert phrasing
Do we need to adjust the alert phrasing when Constant Size checkbox is checked?

**Current Examples from Design Doc:**
**Note:** For slider/number spinner interactions, typically the object response (the new current value) is delivered first (as part of `aria-valuetext`), and then immediately followed by a context response as an alert from a changed `ARIA live region`, a change in the surrounding context.
**When increasing the mass of a mass sphere from 2 to 3 billion kilogram:
- With Constant Size Not checked:
  - **Object response:** {{3}} billion kilograms 
  - **Context response:** As {{mass gets bigger}} force arrows {{get bigger}}, forces now {{force-value-newtons Newtons}}.
- With Constant Size checked:
  - **Object response:** {{3}} billion kilograms 
  - **Context response:** As {{density increases}} force arrows {{get bigger}}, forces now {{force-value-newtons Newtons}}.

Decreasing mass of a mass sphere from 2 to 1 billion kilogram:
- With Constant Size Not checked
  - **Object response:** {{1}} billion kilograms
  - **Context response:** As {{mass gets smaller}} force arrows {{get smaller}}, forces now {{force-value-newtons Newtons}}.
- With Constant Size checked:
  - **Object response:** {{1}} billion kilograms 
  - **Context response:** As {{density decreases}} force arrows {{get smaller}}, forces now {{force-value-newtons Newtons}}.

@arouinfar, are you suggesting any changes to the context responses (the alert phrasing), or just the help text?

**Comment by terracoda** [comment]
@arouinfar, if you would rather meet, just let me know. Once you have commented, please re-assign me ;-)

**Comment by terracoda** [comment]
@arouinfar, I most concerned about this comment:
> I think changing the descriptions to refer to density adds on the layer of complexity we were trying to eliminate in the first place.

Note that the students we are trying to reach with the density/force alerts cannot see the color changes. And also note that the alert is delivered together with the changed mass value. This may not have been apparent when using the A11y View. It is much more clear when using an actual screen reader, or looking at the examples that I copied in from the design document.

**Comment by arouinfar** [comment]
@terracoda thank you so much for your detailed response! I think I have much better sense of what users will actually hear while interacting with the sim. I also spent some time today playing with dev.39 using VoiceOver. I'm not savvy enough to figure out how to access the help text and everything in the PDOM, but I do believe I was hearing the static descriptions and interactive alerts, including things very similar to the examples in https://github.com/phetsims/gravity-force-lab-basics/issues/149#issuecomment-502046289.

In context, I think this way of describing things when Constant Size is off does feel really natural:

```
{{1}} billion kilograms
As {{density decreases}} force arrows {{get smaller}}, forces now {{force-value-newtons Newtons}}.
```

> Note that the students we are trying to reach with the density/force alerts cannot see the color changes. And also note that the alert is delivered together with the changed mass value. This may not have been apparent when using the A11y View. It is much more clear when using an actual screen reader, or looking at the examples that I copied in from the design document.

Now that I hear everything together, I take back my previous comment. I don't think referring to the density is adding any complexity here.

However, I am still uneasy with the help text that reads:
`Change {{density of spheres}} in billions of kilograms.`

There's a value-unit mismatch here. You proposed some good alternatives in https://github.com/phetsims/gravity-force-lab-basics/issues/149#issuecomment-502039410, and I think option (4) is probably the way to go. 
>Density and general volume when Constant Size checked:
> * Constant Size is NOT checked: Change {{mass of spheres}} in billions of kilograms.
> * Constant Size is checked: Change {{density of spheres}} in billions of kilograms {{per unit volume}}.
 
I think referring to density in the help text better differentiates between the Constant Size modes and has a nice parallel to the descriptions/alerts associated with changing the mass. I don't think it's particularly important to be precise with the density units, so I would use the more generalized "per unit volume" for GFL and GFLB.

**Comment by terracoda** [comment]
@arouinfar, thanks for your review and feedback. I'll make the changes clear in the design doc and assign them over to @zepumph in the morning!

**Comment by terracoda** [comment]
@zepumph, in the end of this discussion we have one small string that needs to be added to the help text for the mass controls when the Constant Size checkbox is checked.  There isno change to the help text when the Constant Size checkbox is not checked.

### The New Additional String:
Please add the **bolded** string  to the end of the help text when the Constant Size check box is checked.
- Constant Size is NOT checked: Change {{mass of spheres}} in billions of kilograms.
- Constant Size is checked: Change {{density of spheres}} in billions of kilograms **{{per unit volume}}**.

**Comment by zepumph** [comment]
Sounds good. Anything else for this issue?

**Comment by terracoda** [comment]
@zepumph, I went to the commit, it looks like the change has been made in Gravity Force Lab **regular,** but I don't see it yet in Gravity Force Lab Basics where we have "billions" of kilograms.
<img width="791" alt="Screen Shot 2019-06-18 at 11 32 48 AM" src="https://user-images.githubusercontent.com/1902116/59671464-fbbfac80-91bd-11e9-849b-d438219877e0.png">

It will good to have both sims do the same thing, i.e., add {{per unit volume}} at the end of the help text that contains "density", but in BASICS we need additionally "billions of kilograms".

**Comment by zepumph** [comment]
Sorry I didn't check my work thoroughly. I noticed while making this new change that these BASICS strings were in the REGULAR strings file. I moved them to GFLB for clarity. Back to you for a sanity check.

**Comment by terracoda** [comment]
Thanks! All looking good now when Constant Size is checked!
Closing.
